overlord/assertstate/assertstate.go

```diff
@@ -326,15 +326,18 @@ func AutoAliases(s *state.State, info *snap.Info) (map[string]string, error) {
 func delayedCrossMgrInit() {
 	// hook validation of refreshes into snapstate logic
 	snapstate.ValidateRefreshes = ValidateRefreshes
-	// hook auto refresh of assertions into snapstate
+	// hook auto refresh of assertions (snap declarations) into snapstate
 	snapstate.AutoRefreshAssertions = AutoRefreshAssertions
 	// hook retrieving auto-aliases into snapstate logic
 	snapstate.AutoAliases = AutoAliases
 }
 
 // AutoRefreshAssertions tries to refresh all assertions
 func AutoRefreshAssertions(s *state.State, userID int) error {
-	return RefreshSnapDeclarations(s, userID)
+	if err := RefreshSnapDeclarations(s, userID); err != nil {
+		return err
+	}
+	return RefreshValidationSetAssertions(s, userID)
```

Comment:
thanks, this looks right but there's a missing bit that we forgot to consider,  we need to update the relevant ValidationSetTracking.Current I think?  (for the enforce case that will be more complicated because we need to update it only if we can take the new state)

Reply:
Done.